egui_for_winit/src/lib.rs

```diff
@@ -0,0 +1,638 @@
+    window.scale_factor() as f32
+}
+
+pub fn screen_size_in_pixels(window: &winit::window::Window) -> egui::Vec2 {
+    // let (width_in_pixels, height_in_pixels) = display.get_framebuffer_dimensions();
+    // egui::vec2(width_in_pixels as f32, height_in_pixels as f32)
+    let size = window.inner_size().to_logical(window.scale_factor());
+    egui::vec2(size.width, size.height)
+}
+
+/// Handles the integration between egui and winit.
+pub struct State {
+    start_time: std::time::Instant,
+    egui_input: egui::RawInput,
+    pointer_pos_in_points: Option<egui::Pos2>,
+    any_pointer_button_down: bool,
+    current_cursor_icon: egui::CursorIcon,
+    /// What egui uses.
+    current_pixels_per_point: f32,
+
+    screen_reader: crate::screen_reader::ScreenReader,
+
+    #[cfg(feature = "copypasta")]
+    clipboard: Option<copypasta::ClipboardContext>,
+}
+
+impl State {
+    /// Initialize with the native pixels_per_point (dpi scaling).
+    pub fn new(window: &winit::window::Window) -> Self {
+        Self::from_pixels_per_point(crate::native_pixels_per_point(window))
+    }
+
+    /// Initialize with a given dpi scaling.
+    pub fn from_pixels_per_point(pixels_per_point: f32) -> Self {
+        Self {
+            start_time: std::time::Instant::now(),
+            egui_input: egui::RawInput {
+                pixels_per_point: Some(pixels_per_point),
+                ..Default::default()
+            },
+            pointer_pos_in_points: None,
+            any_pointer_button_down: false,
+            current_cursor_icon: egui::CursorIcon::Default,
+            current_pixels_per_point: pixels_per_point,
+
+            screen_reader: crate::screen_reader::ScreenReader::default(),
+
+            #[cfg(feature = "copypasta")]
+            clipboard: init_clipboard(),
+        }
+    }
+
+    /// The same as what egui uses.
+    pub fn pixels_per_point(&self) -> f32 {
+        self.current_pixels_per_point
+    }
+
+    /// The current input state.
+    /// This is changed by [`Self::on_event`] and cleared by [`Self::take_egui_input`].
+    pub fn egui_input(&self) -> &egui::RawInput {
+        &self.egui_input
+    }
+
+    /// Prepare for a new frame by extracting the accumulated input.
+    pub fn take_egui_input(&mut self, display: &winit::window::Window) -> egui::RawInput {
+        let pixels_per_point = self.pixels_per_point();
+
+        self.egui_input.time = Some(self.start_time.elapsed().as_secs_f64());
+
+        // On Windows, a minimized window will have 0 width and height.
+        // See: https://github.com/rust-windowing/winit/issues/208
+        // This solves an issue where egui window positions would be changed when minimizing on Windows.
+        let screen_size = screen_size_in_pixels(display);
+        self.egui_input.screen_rect = if screen_size.x > 0.0 && screen_size.y > 0.0 {
+            Some(egui::Rect::from_min_size(
+                Default::default(),
+                screen_size / pixels_per_point,
```

Comment:
screen_size is already in logical space. It should probably not be divided again.

Reply:
This could potentially be the reason for the resizing issues I mentioned earlier, you can see it doesn't quite reach the edge of the screen. I think I had the DPI set to below the default when that happened

I never tested with Glium, so it could have also been an issue with my rebasing or code who knows

Reply:
Oops - there was a mistaken call to `to_logical` in `screen_size_in_pixels`. Fixed!